src/main/java/ch/wisv/areafiftylan/service/TeamServiceImpl.java

```diff
@@ -89,8 +88,10 @@ public Team update(Long teamId, TeamDTO input) {
         }
 
         // If the Captain username is set, change the Captain
-        if (!Strings.isNullOrEmpty(input.getCaptainUsername())) {
-            User captain = userService.getUserByUsername(input.getCaptainUsername()).get();
+        String captainUsername = input.getCaptainUsername();
+        if (!Strings.isNullOrEmpty(captainUsername)) {
```

Comment:
I would also perform a check if the captain in input is that same as the current captain. I know this doesn't change anything in functionality but I think it's better code. Less execution.